### ISSUE TITLE
Image.frombytes  TypeError for JPEG decoder invocation

### ISSUE THREAD

**Issue by mobilistix** [python-pillow/Pillow#1382]
Configuration:

python 3.4, Ubuntu 14.04, Pillow 2.9.0
libjpeg8-dev:amd64 (8c-2ubuntu8) 

Here is test code:

from PIL import Image
im = Image.open('/tmp/test.jpg')
print(im.format, im.size, im.mode)
im.close()

file_ = open('/tmp/test.jpg','rb')
b = file_.read()
file_.close()
print('read file with ',len(b), ' bytes')

decoder = "jpeg"
mode = "RGB"
im_size = 320, 240
image = Image.frombytes(mode, im_size, b, decoder, None)

print('done')

Output

JPEG (320, 240) RGB
read file with  16143  bytes
Traceback (most recent call last):
   File "piltest.py", line 17, in <module>
     image = Image.frombytes(mode, im_size, b, decoder, None)
   File "/root/.virtualenvs/cv/lib/python3.4/site-packages/PIL/Image.py", line 2085, in frombytes
     im.frombytes(data, decoder_name, args)
   File "/root/.virtualenvs/cv/lib/python3.4/site-packages/PIL/Image.py", line 741, in frombytes
     d = _getdecoder(self.mode, decoder_name, args)
   File "/root/.virtualenvs/cv/lib/python3.4/site-packages/PIL/Image.py", line 418, in _getdecoder
     return decoder(mode, *args + extra)
 TypeError: function takes at least 3 arguments (2 given)

**Comment by radarhere** [comment]
This will be a start at least - for the jpeg decoder, there are two arguments required - `rawmode`, what you'd like back from the decoder, and `jpegmode`, what is in the file. You can provide `None` for `jpegmode`, but you need to provide a rawmode.

So you need to amend your command to `Image.frombytes(mode, im_size, b, decoder, rawmode, jpegmode)`

**Comment by radarhere** [comment]
Can I also ask, what are you trying to achieve? I'm wondering if there's a simpler way to make it to your end goal.

**Comment by mobilistix** [comment]
Thanks.  I am updating this python server (https://github.com/ragulin/face-recognition-server) to run under python 3 and latest opencv.  Server receives jpeg frames from websocket in byte buffer.

Original code is:  image = Image.open(StringIO.StringIO(message)).
I replaced with: image = Image.open(BytesIO(message)) which works,
Along the way I tried the frombytes approach above.

Thank you for clarifying the frombytes args.
